ckanext/datapusher/logic/action.py

```diff
@@ -84,18 +84,33 @@ def datapusher_submit(context, data_dict):
         'error': '{}',
     }
     try:
-        task_id = p.toolkit.get_action('task_status_show')(context, {
+        existing_task = p.toolkit.get_action('task_status_show')(context, {
             'entity_id': res_id,
             'task_type': 'datapusher',
             'key': 'datapusher'
-        })['id']
-        task['id'] = task_id
+        })
+
+        if existing_task and existing_task.get('state') == 'pending':
+            updated = datetime.datetime.strptime(
+                existing_task['last_updated'], '%Y-%m-%dT%H:%M:%S.%f')
+            time_since_last_updated = datetime.datetime.now() - updated
+            if time_since_last_updated > datetime.timedelta(days=1):
+                # it's been a while since the job was last updated - it's more
+                # likely something went wrong with it and the state wasn't
+                # updated than its still in progress. Let it be restarted.
```

Comment:
A day seems like a very long time. Maybe make this shorter and use a nice descriptive constant name like `ASSUME_TASK_STALE_AFTER = timedelta(..)`